snuba/datasets/outcomes.py

```diff
@@ -0,0 +1,173 @@
+from datetime import datetime
+import jsonschema
+import six
+
+from snuba import schemas, settings
+from snuba.clickhouse import *
+from snuba.datasets import DataSet
+from snuba.processor import Processor, InvalidMessage
+from snuba.writer import row_from_processed_event
+
+class OutcomesTableSchema(TableSchema):
+    def __init__(self, *args, **kwargs):
+        super(OutcomesTableSchema, self).__init__(*args, **kwargs)
+
+        self.CLICKHOUSE_CLUSTER = None
+        self.DATABASE = 'default'
+        self.LOCAL_TABLE = 'outcomes_local'
+        self.DIST_TABLE = 'outcomes_dist'
+        self.QUERY_TABLE = self.DIST_TABLE # For prod, queries are run against the dist table
+        self.CAN_DROP = False
+
+        # The sort key starts with timestamp, as this dataset will
+        # be used primarily by TSDB, so all queries should be
+        # within a time range, but using hourly granularity for
+        # the sort so that within a given time bucket we still
+        # have decent size chunks by org and project
+        self.ORDER_BY = '(toStartOfHour(timestamp), org_id, project_id)'
+        self.PARTITION_BY = 'toMonday(timestamp)'
+        self.SHARDING_KEY = 'cityHash64(timestamp)'
```

Comment:
Is there a reason behind sharding outcomes by timestamp while we shard events by event_id ?

Reply:
We aren't guaranteed to have an event_id in this dataset, and I just needed something that would be relatively uniform. We could always just do `random()` but that makes me vaguely uncomfotrable because we would get different/non-deterministic shards if we were to try and rebuild the table from the source data.